### PR TITLE
Custom partition key name

### PR DESCRIPTION
**Pull request by kxmas**
I use snake case for json serialization, but the @PartitionKey annotation only uses the java field name. This PR addresses that mismatch.

```
@PartitionKey("partition_key")
String partitionKey;
```

### CODE REVIEW

**Comment by kxmas** [comment]
> Codacy Here is an overview of what got changed by this pull request:
> 
> ```diff
> Issues
> ======
> - Added 2
>            
> ```
> 
> See the complete overview on [Codacy](https://app.codacy.com/app/Microsoft/spring-data-cosmosdb/pullRequest?prid=3753899&bid=13180530)

I am unsure as how to satisfy these two Codacy demands. It's complaining about the two small entity classes that I added to the DocumentDbEntityInformationUnitTest in order to properly test the enhancement.

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/DocumentDbEntityInformationUnitTest.java [hunk 1]
```diff
@@ -60,9 +61,50 @@ public void testCustomCollectionName() {
...
+    @Document
+    private class VolunteerWithCustomPartitionKey {
+        private String id;
+        @PartitionKey("vol_name")
+        private String name;
+    }
+
+    @Document
+    private class VolunteerWithPartitionKey {
+        private String id;
```
[the remark this answers is not in the document]
Comment (Incarnation-p-lee) [reply src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/DocumentDbEntityInformationUnitTest.java]: you can add `Getter` and `Setter` for this field.

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/DocumentDbEntityInformationUnitTest.java [hunk 2]
```diff
@@ -60,9 +61,50 @@ public void testCustomCollectionName() {
...
+        @PartitionKey
+        private String name;
```
[the remark this answers is not in the document]
Comment (Incarnation-p-lee) [reply src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/DocumentDbEntityInformationUnitTest.java]: the same as below.

**Comment by kxmas** [comment]
@Incarnation-p-lee Adding an unused getter and setter fixed the unused field issue. Thanks.

**Comment by Incarnation-p-lee** [comment]
@kxmas
Great, seems the travis CI has some network issue for integration test. I will trigger the build later.

**Comment by kxmas** [comment]
> @kxmas
> Great, seems the travis CI has some network issue for integration test. I will trigger the build later.

The Travis CI build pipeline appears to be broken. I didn't look too hard, but I can't find a PR where it succeeded.

**Comment by kushagraThapar** [comment]
@kxmas can you please resolve the conflicts, so I can run additional CI to make sure we are good with all the tests.

**Comment by kxmas** [comment]
Done

**Pull request merged**